app/models/booking.rb

```diff
@@ -2,5 +2,7 @@ class Booking < ApplicationRecord
   belongs_to :user
   belongs_to :ship
 
-  validates :user_id, :ship_id, presence: true, numericality: { only_integer: true }
+  validates :user, :ship, presence: true
+  validates :date_start, :date_end, presence: true, if: -> {:date_start < :date_end} 
```

Comment:
Not 100% sure about that `if: -> {:date_start < :date_end}`

Reply:
Je crois que ça passera pas
Je propose ça : 
```
validate :end_after_start
private
def end_after_start
  return if end_date.blank? || start_date.blank?
 
  if end_date < start_date
    errors.add(:end_date, "must be after the start date") 
  end 
 end
```

[Rails_blog_ref](https://myrailslearnings.wordpress.com/2013/09/10/how-to-validate-end-date-after-start-date/)